DEV_GUIDE.md

```diff
@@ -0,0 +1,102 @@
+
+> :warning: **If you are using Podman please see the notes below**
+> 
+## JDK target
+
+The project targets language Java level 17.
+
+## Building / Running the Tests
+
+Build the project like this:
+
+```shell
+$ mvn clean install
+```
+
+The running of the tests can be controlled with the following Maven properties:
+
+| property           | description                                                                               |
+|--------------------|-------------------------------------------------------------------------------------------|
+| `-DskipUTs=true`   | skip unit tests                                                                           |
+| `-DskipITs=true`   | skip integration tests                                                                    |
+| `-DskipTests=true` | skip all tests                                                                            |
+| `-Pdebug`          | enables logging so you can see what the Kafka clients, Proxy and in VM brokers are up to. |
+
+The kafka environment used by the integrations tests can be _defaulted_ with these two environment variables.
+
+| env var                       | default | description                                                                                                                             |
+|-------------------------------|---------|-----------------------------------------------------------------------------------------------------------------------------------------|
+| `TEST_CLUSTER_EXECUTION_MODE` | `IN_VM` | `IN_VM` or `CONTAINER`. if `IN_VM`, kafka will be run same virtual machines as the integration test. Otherwise containers will be used. |
+| `TEST_CLUSTER_KRAFT_MODE`     | `true`  | if true, kafka will be run in kraft mode.                                                                                               |
+
+When the integration-tests are run in `CONTAINER` mode, the kafka/zookeeper logs are written to a location specified by
+the `container.logs.dir`  system property. When run through Maven this is defaulted to `integrationtests/target/container-logs`.
+
+Pass the `-Dquick` option to skip all tests and non-essential plug-ins and create the output artifact as quickly as possible:
+
+```shell
+$ mvn clean verify -Dquick
+```
+
+### Code Formatting
+
+Run the following command to format the source code and organize the imports as per the project's conventions:
+
+```shell
+$ mvn process-sources
+```
+
+## Running Integration Tests on Podman
+
+There is an incompatibility between HTTP connection timeout expectations of 
+[testcontainers-java](https://github.com/testcontainers/testcontainers-java) and the Podman API. This
+can result in sporadic test failures when running the Integration Tests under Podman.  It manifests as
+failed or hanging REST API calls that lead to test failures and test hangs.
+
+It affects Linux and Mac OS X.
+
```

Comment:
I think we need to give people some clear guidance around identifying the issue.
```suggestion
It affects Linux and Mac OS X.
On Linux it manifests as Http calls failing with a `Broken Pipe` exception. 
Similarly on MacOS we see a `localhost:XXX failed to respond`.
```